src/core/window.js

```diff
@@ -1,6 +1,8 @@
 function makeWindow() {
   var win = {
-    location: {},
+    location: {
+      href: "https://app.swaggerhub.com/apis/smartbear/petstore/1.0.0##/pet/addPet"
```

Comment:
Added this for unit testing servers with relative urls

Reply:
I hope this is just for draft purposes. You should be able to define a new window with custom data in Mocha.

Reply:
This was the wrong approach to what I was trying to achieve, so probably better for to explain motivation...
In purpose of `buildUrl()`:
1. If URL is absolute, all is good; return the URL
2. If URL is relative, then we want the base to be the Server URL
3. However, if the Server URL itself is relative, we want the base to be the location of Definition YAML

In case #3 I was (incorrectly) setting the base URL to window location

Is there a way to get the location of the Definition YAML?
– I know what it is for our system, but it need to ben generic for Swagger-UI's system

Reply:
> Is there a way to get the location of the Definition YAML?

e.g. in petstore, you enter an external url path of a definition in the topbar? in that case, you can use `spec.url`